src/status_im/contexts/wallet/sheets/slippage_settings/view.cljs

```diff
@@ -37,10 +41,26 @@
   []
   (let [current-slippage                (rf/sub [:wallet/swap-max-slippage])
         account-color                   (rf/sub [:wallet/current-viewing-account-color])
+        asset-to-receive                (rf/sub [:wallet/swap-asset-to-receive])
+        amount-out                      (rf/sub [:wallet/swap-proposal-amount-out])
         [max-slippage set-max-slippage] (rn/use-state (str current-slippage))
         [error set-error]               (rn/use-state nil)
         [custom? set-custom?]           (rn/use-state (not-any? #{current-slippage}
                                                                 constants/slippages))
+        receive-token-decimals          (:decimals asset-to-receive)
+        receive-token-symbol            (:symbol asset-to-receive)
+        amount-out-whole-number         (number/hex->whole amount-out receive-token-decimals)
+        amount-out-num                  (number/to-fixed amount-out-whole-number
+                                                         (min constants/min-token-decimals-to-display
+                                                              receive-token-decimals))
+        slippage-num                    (-> (money/bignumber amount-out-num)
+                                            (money/mul (money/bignumber max-slippage))
+                                            (money/div (money/bignumber 100)))
+        amount-out-minus-slippage       (money/sub (money/bignumber amount-out-num) slippage-num)
+        receive-amount                  (if (money/greater-than amount-out-minus-slippage
+                                                                (money/bignumber 0))
+                                          (str amount-out-minus-slippage)
+                                          0)
```

Comment:
Might be better to have function definitions outside the component for these, which will also allow to thread some let assignments that are only used once. We should also consider whether this belongs in a re-frame subscription (cc: @ilmotta)

Reply:
Thanks for tagging me @clauxx. I think the code is a bit wild guys. With this amount of let bindings almost no human can decipher the dependencies between bindings without being very attentive. Another thing that complicates the readability is that the bindings are mixed with other concerns for local state, subs and events.

One suggestion (untested) is to extract the calculation to a separate function. Could be a sub in theory as @clauxx suggests. Note that I had to manually unwind the dependencies between the bindings to see what depends on what and this is time consuming.

The other important factor is that we should code to support REPL usage. Long calculations, tied to view concerns make it quite hard to verify how the calculation works. The other main benefit from @clauxx suggestion is that now we can test this implementation in a fast unit test, throw at it many inputs and check and be confident it works.

```clojure
(defn- calculate-receive-amount
  [asset-to-receive amount-out max-slippage]
  (let [receive-token-decimals    (:decimals asset-to-receive)
        amount-out-whole-number   (number/hex->whole amount-out receive-token-decimals)
        amount-out-num            (number/to-fixed amount-out-whole-number
                                                   (min constants/min-token-decimals-to-display
                                                        receive-token-decimals))
        slippage-num              (-> (money/bignumber amount-out-num)
                                      (money/mul (money/bignumber max-slippage))
                                      (money/div (money/bignumber 100)))
        amount-out-minus-slippage (money/sub (money/bignumber amount-out-num) slippage-num)
        receive-amount            (if (money/greater-than amount-out-minus-slippage
                                                          (money/bignumber 0))
                                    (str amount-out-minus-slippage)
                                    0)]
    receive-amount))
```

And then replace in the view component

```diff
@@ -47,20 +65,10 @@
         [error set-error]               (rn/use-state nil)
         [custom? set-custom?]           (rn/use-state (not-any? #{current-slippage}
                                                                 constants/slippages))
-        receive-token-decimals          (:decimals asset-to-receive)
         receive-token-symbol            (:symbol asset-to-receive)
-        amount-out-whole-number         (number/hex->whole amount-out receive-token-decimals)
-        amount-out-num                  (number/to-fixed amount-out-whole-number
-                                                         (min constants/min-token-decimals-to-display
-                                                              receive-token-decimals))
-        slippage-num                    (-> (money/bignumber amount-out-num)
-                                            (money/mul (money/bignumber max-slippage))
-                                            (money/div (money/bignumber 100)))
-        amount-out-minus-slippage       (money/sub (money/bignumber amount-out-num) slippage-num)
-        receive-amount                  (if (money/greater-than amount-out-minus-slippage
-                                                                (money/bignumber 0))
-                                          (str amount-out-minus-slippage)
-                                          0)
+        receive-amount                  (calculate-receive-amount asset-to-receive
+                                                                  amount-out
+                                                                  max-slippage)
```

Reply:
Thanks for the suggestions, just updated the code. I totally agree with you, tbh, I just rushed things for the sake of agility, and this is the result :D.

Will do a follow up with some tests. Ideally I should do it in the same PR, but want to merge this so we can get a release candidate for swaps as soon as possible.